File: app/src/main/java/com/yd/xzqcomponentkt/welcome/WelcomeActivity.java
package com.yd.xzqcomponentkt.welcome;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import com.yd.xzqcomponentkt.MainActivity;


public class WelcomeActivity extends AppCompatActivity implements Runnable {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().getDecorView().postDelayed(this, 2000);
    }

    @Override
    public void run() {
        if (isFinishing()) {
            finish();
            return;
        }
        MainActivity.start(this);
        finish();
    }
}

File: app/src/main/java/com/yd/xzqcomponentkt/welcome/WelcomeActivity.kt
package com.yd.xzqcomponentkt.welcome

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.yd.xzqcomponentkt.MainActivity


class WelcomeActivity : AppCompatActivity(), Runnable {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.decorView.postDelayed(this, 2000)
    }

    override fun run() {
        if (isFinishing) {
            finish()
            return
        }
        MainActivity.start(this)
        finish()
    }
}
